components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/UserStoreAttributeMappingParser.java

```diff
@@ -0,0 +1,178 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.user.store.configuration.utils;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.user.store.configuration.model.ChangedUserStoreAttributeDO;
+import org.wso2.carbon.identity.user.store.configuration.model.UserStoreAttributeDO;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.xml.stream.XMLStreamException;
+
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.AD_ATTRIBUTE_MAPPING_CONFIG;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.ATTRIBUTES_DIR;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.ATTRIBUTE_ID;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.CLAIM_URI;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DISPLAY_NAME;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.LDAP_ATTRIBUTE_MAPPING_CONFIG;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.OPERATION;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.USERSTORE_DIR;
+
+/**
+ * Parser to read and build user store mappings attribute changes.
+ */
+public class UserStoreAttributeMappingParser {
+
+    private static Map<String, ChangedUserStoreAttributeDO> adUserStoreAttrChangeDOMap;
+    private static Map<String, ChangedUserStoreAttributeDO> ldapUserStoreAttrChangeDOMap;
+    private static final Log LOG = LogFactory.getLog(UserStoreAttributeMappingParser.class);
+
+    private UserStoreAttributeMappingParser() {
+
+        init();
+    }
+
+    private static final class ParserHolder {
+
+        static final UserStoreAttributeMappingParser PARSER = new UserStoreAttributeMappingParser();
+    }
+
+    public static UserStoreAttributeMappingParser getInstance() {
+
+        return ParserHolder.PARSER;
+    }
+
+    private static void init() {
+
+        ldapUserStoreAttrChangeDOMap = readChangeFiles(LDAP_ATTRIBUTE_MAPPING_CONFIG);
+        adUserStoreAttrChangeDOMap = readChangeFiles(AD_ATTRIBUTE_MAPPING_CONFIG);
+    }
+
+    private static Map<String, ChangedUserStoreAttributeDO> readChangeFiles(String attrMappingsFileName) {
+
+        InputStream inStream = null;
```

Comment:
Check the above comments and do the necessary changes in all the places.

Reply:
fixed c9f41a6f5c0044ce53b9f60f6df2b67f0bc6d2d8